### PR TITLE
Add Error test cases and simplify duration parsing logic

### PR DESCRIPTION
**Pull request by thomshutt**

### CODE REVIEW

File: mpd/duration.go [hunk 1]
```diff
@@ -141,77 +143,61 @@ func fmtInt(buf []byte, v uint64) int {
...
-		minus = true
-		offset++
+	if strings.Contains(str, "-") {
+		return 0, errors.New("Duration cannot be negative")
 	}
 
-	if str[offset] != 'P' {
-		return 0, errors.New("input duration does not have a valid prefix")
+	// Check that only the parts we expect exist and that everything's in the correct order
+	r := regexp.MustCompile(
```
Comment (stuarthicks) [inline mpd/duration.go]: Can we move the regex initialisation outside the method? Compiling the regex on every invocation feels unnecessary.

File: mpd/duration.go [hunk 2]
```diff
@@ -141,77 +143,61 @@ func fmtInt(buf []byte, v uint64) int {
...
+		"^P" + // Must start with a 'P'
+		"(\\d+D)?" + // We only allow Days for durations, not Months or Years
```
Comment (stuarthicks) [inline mpd/duration.go]: Can we use named captures? I prefer not to read captures out by slice index.

Comment (thomshutt) [reply mpd/duration.go]: Go doesn't really support named captures without doing a double loop of an array of parameter names and one of parameters

File: mpd/duration.go (hunk 1 above)
Comment (thomshutt) [reply mpd/duration.go]: Yep, done!

**Pull request merged**